### ISSUE TITLE
`--no-brew-update` overrides `--all`

### ISSUE THREAD

**Issue by wickles** [buo/homebrew-cask-upgrade#238]
```
❯ brew cu --report-only --quiet --all
==> Updating Homebrew
Already up-to-date.
==> Finding outdated apps
==> Found outdated apps
       Cask                Current           Latest            A/U    Result    
...
 6/15  firefox             108.0             117.0.1            Y   [ FORCED ]  
 7/15  obsidian            1.4.11            1.4.12             Y   [ FORCED ]  
 8/15  onedrive            22.141.0703.0002  23.153.0724.0003   Y   [ FORCED ]  
 9/15  ppsspp              1.16              1.16.1                 [OUTDATED]  
10/15  spotify             1.2.10.760        1.2.19.941         Y   [ FORCED ]  
...
```

```
❯ brew cu --report-only --quiet --all --no-brew-update
==> Finding outdated apps
==> Found outdated apps
     Cask    Current  Latest  A/U    Result    
1/1  ppsspp  1.16     1.16.1       [OUTDATED]  
```

Not sure how this is happening. Only difference is:

https://github.com/buo/homebrew-cask-upgrade/blob/9cfd04381733b6d39c35cfbd95b382fb2c533376/lib/bcu/command/upgrade.rb#L28

**Comment by wickles** [comment]
Personally I would prefer `brew cu` to inherit from environment variables `HOMEBREW_NO_AUTO_UPDATE` and `HOMEBREW_AUTO_UPDATE_SECS`.

**Comment by ondrejfuhrer** [comment]
> Personally I would prefer brew cu to inherit from environment variables HOMEBREW_NO_AUTO_UPDATE and HOMEBREW_AUTO_UPDATE_SECS as documented in man brew.

I agree, that's a valid feature request. Maybe we could separate that to a separate issue and let the community build that 🙂 The `--no-brew-update` was built before those options were present so it's just a historical reason why this is not yet respecting it.

**Comment by wickles** [comment]
> However I can't seem to be able to reproduce that.

Appears to be related to another configuration flag I use. Try setting environment variable `HOMEBREW_NO_INSTALL_FROM_API=1`. When I unset that they produce the same output. However we should prefer to get the same output either way.

**Comment by ondrejfuhrer** [comment]
> HOMEBREW_NO_INSTALL_FROM_API

We don't actually do anything with this flag and it seems that it's actually a development command so therefore not really to be used by "general public". The issue is that when the flag is set, then `Cask::version` doesn't report installed version for most of the casks and therefore we don't know that they are outdated. I'm not sure if it's worth trying to support and try to find a different way for getting current version then this official API.
